src/com/owncloud/android/files/FileMenuFilter.java

```diff
@@ -215,6 +215,20 @@ private void filter(List<Integer> toShow, List <Integer> toHide) {
             toShow.add(R.id.action_send_file);
         }
 
+        // FAVORITES
+        if (mFile == null || downloading || uploading || mFile.isFolder() || mFile.keepInSync()) {
```

Comment:
That's something to consider, but right now we have no idea when the integration of the server-side stars will be prioritized (could be soon, could be in two years). On the while I'd prefer the code is as easy to read as possible, a bit of internal coherence will help.